launch_testing_ros/launch_testing_ros/wait_for_topics.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2021 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from threading import Event
+from threading import Lock
+from threading import Thread
+
+import rclpy
+from rclpy.executors import SingleThreadedExecutor
+from rclpy.node import Node
+
+
+class WaitForTopics:
+    """Wait to receive messages on supplied nodes."""
+
+    def __init__(self, topic_tuples, timeout=5.0):
+        self.__ros_context = None
+        self.__ros_node = None
+        self.__ros_executor = None
+        self.topic_tuples = topic_tuples
+        self.timeout = timeout
+        self.start()
+
+    def start(self):
+        self.__ros_context = rclpy.Context()
+        rclpy.init(context=self.__ros_context)
+        self.__ros_executor = SingleThreadedExecutor(context=self.__ros_context)
+        self.__ros_node = MakeTestNode(name='test_node', node_context=self.__ros_context)
+        self.__ros_executor.add_node(self.__ros_node)
+
+        # Start spinning
+        self.__running = True
+        self.__ros_spin_thread = Thread(target=self.spin_function)
+        self.__ros_spin_thread.start()
+
+    def spin_function(self):
+        while self.__running:
+            self.__ros_executor.spin_once(1.0)
+
+    def wait(self):
+        self.__ros_node.start_subscribers(self.topic_tuples)
+        return self.__ros_node.msg_event_object.wait(self.timeout)
+
+    def shutdown(self):
+        self.__running = False
+        self.__ros_spin_thread.join()
+        self.__ros_node.destroy_node()
+        rclpy.shutdown(context=self.__ros_context)
+
+    def __enter__(self):
+        self.wait()
+        return self
+
+    def __exit__(self, exep_type, exep_value, trace):
+        if exep_type is not None:
+            raise Exception('Exception occured, value: ', exep_value)
+        self.shutdown()
+
+
+class MakeTestNode(Node):
+    """Mock node to be used for listening on topics."""
+
+    def __init__(self, name='test_node', node_context=None):
+        super().__init__(node_name=name, context=node_context)
+        self.msg_event_object = Event()
+        self.logger = rclpy.logging.get_logger('wait_for_topics')
```

Comment:
We can use the Node's logger instead
```suggestion
        self.logger = self.get_logger()
```

or just replace any instance of `self.logger` with `self.get_logger()`.